internal/models/namespace.go

```diff
@@ -0,0 +1,20 @@
+package models
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NamespaceDto struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+func (workspace *NamespaceDto) ObjectMetaFromNamespaceDto() metav1.ObjectMeta {
+	if workspace.Namespace == "" {
+		workspace.Namespace = "default"
+	}
+	return metav1.ObjectMeta{
+		Name:      workspace.Name,
+		Namespace: workspace.Namespace,
+	}
+}
```

Comment:
Resources name should use a unique name, they can't all share the same name.

Reply:
> ## Naming convention
> ### k8s not k8
> 🔥 Please rename all instances of `k8` to `k8s`, like `k8Client` to `k8sClient`
> 
> ### Constants
> Regarding constants in `*_handler.go`:
> 
> ```go
> const (
>     NODE_NAME_KEYWORD = "name"
>     NAMESPACE_KEYWORD = "namespace"
>     DEFAULT_NAMESPACE = "default"
>  )
> ```
> 
> Golang convention is to use CamelCase for constants https://stackoverflow.com/questions/22688906/go-naming-conventions-for-const
> 
> ### Namespace name confusion
> The embedded dto.Namespace struct can be renamed to `dto.Metadata`, because `namespace.name` and `namespace.namspace` is so confusing, metadata is used by Kubernetes https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#describing-a-kubernetes-object

Got it thanks for explanation , 
> Regarding constants in `*_handler.go`:
I was thinking of creating  file for such constants in k8s pkg and use it in all handlers to get all these constants 
instead of keep defining them in all files. similar to the status file in http pkg of (go src) eg `http.StatusOK`